File: 1-easy/problems/1437-check-if-all-1s-are-at-least-length-k-places-away/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def kLengthApart(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        prev = -k-1
        for i in range(len(nums)):
            if not nums[i]:
                continue
            if i-prev <= k:
                return False
            prev = i
        return True

File: 1-easy/problems/1437-check-if-all-1s-are-at-least-length-k-places-away/solution.ts
function kLengthApart(nums: any, k: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :type k: int
  :rtype: bool
  """
  prev = -k-1
  for (i in Array.from({length: (nums}, (_, i) => i).length)) {
      if (not nums[i]) {
          continue
      if (i-prev <= k) {
          return false;
      prev = i
  return true;
}

export { kLengthApart };
